Account/FirefoxAccount.swift

```diff
@@ -197,67 +198,52 @@ open class FirefoxAccount {
     }
     
     public class FxAProfile {
-        open var avatarUrl: URL?
-        open var avatarImageData: Data?
         open var displayName: String?
         open let email: String
+        open var avatar: Avatar?
         
         init(email: String, displayName: String?, avatar: String?) {
             self.email = email
             self.displayName = displayName
-            self.avatarImageData = nil
 
             if let avatarStr = avatar {
-                self.avatarUrl = URL(string: avatarStr)
-            } else {
-                self.avatarUrl = nil
-            }            
+                self.avatar = Avatar(url: URL(string: avatarStr))
```

Comment:
Given that `URL` is failable, you might as well make `avatar` non-optional and make it `let`, and do:

```
self.avatar = Avatar(url: avatarStr.asURL)
```